pyfstat/helper_functions.py

```diff
@@ -246,59 +247,55 @@ def get_comb_values(F0, frequencies, twoF, period, N=4):
     return comb_frequencies, twoF[comb_idxs], freq_err * np.ones(len(comb_idxs))
 
 
-def run_commandline(cl, raise_error=True, return_output=True):
-    """Run a string cmd as a subprocess, check for errors and return output.
+def run_commandline(
+    cl: str, raise_error: bool = True, return_output: bool = False
+) -> Union[subprocess.CompletedProcess, None]:
+    """Run a string command as a subprocess.
 
     Parameters
     ----------
-    cl: str
+    cl:
         Command to run
-    log_level: int
-        Sets the logging level for some of this function's messages.
-        See https://docs.python.org/library/logging.html#logging-levels
-        Default is '20' (INFO).
-        FIXME: Not used for all messages.
-    raise_error: bool
+    raise_error:
         If True, raise an error if the subprocess fails.
-        If False, continue and just return `0`.
-    return_output: bool
-        If True, return the captured output of the subprocess (stdout and stderr).
-        If False, return nothing on successful execution.
+        If False, just log the error, continue, and return ``None``.
+    return_output:
+        If True, return the ``subprocess.CompletedProcess`` object.
+        If False, return ``None``.
 
     Returns
     ----------
-    out: str or int, optional
-        The captured output of the subprocess (stdout and stderr)
-        if `return_output=True`.
-        0 on failed execution if `raise_error=False`.
+    out:
+        The ```subprocess.CompletedProcess`` of the subprocess
+        if ``return_output=True``. ``None`` if ``return_output=False``
+        or on  failed execution if ``raise_error=False``.
     """
 
     logger.info("Now executing: " + cl)
     if "|" in cl:
         logger.warning(
-            "Pipe ('|') found in commandline, errors may not be" " properly caught!"
+            "Pipe ('|') found in commandline, errors may not be  properly caught!"
         )
     try:
+        completed_process = subprocess.run(
+            cl,
+            check=True,
+            shell=True,
+            capture_output=True,
```

Comment:
https://docs.python.org/3/library/subprocess.html -> 
> If you wish to capture and combine both streams into one, use stdout=PIPE and stderr=STDOUT instead of capture_output.

The following seems to restore previous behaviour on capturing XLAL errors (try the example posted in previous comment)
```suggestion
            stdout=subprocess.PIPE,
            stderr=subprocess.STDOUT,
```

However it breaks the test, and I'm not sure if it's the right thing.

Maybe the stderr handling 5 lines from below here just isn't quite right and can be fixed instead?